File: C++/Memory.cpp
#include <bits/stdc++.h>
using namespace std;
// import class Hardware
#include "Hardware.cpp"

class Memory : public Hardware
// kelas yang digunakan untuk mengimplementasikan Memory dan juga turunan dari kelas Hardware
{
private:
    // private atribute dari kelas Memory
    string frequency;
    int memorySize;
    string supportsCuda;

public:
    // konstrutor
    Memory() {}

    // mengeset nilai atribut frequency
    void setFrequency(string frequency)
    {
        this->frequency = frequency;
    }

    // mengembalikan nilai atribut frequency
    string getFrequency()
    {
        return this->frequency;
    }

    // mengeset nilai atribut memorySize
    void setMemorySize(int memorySize)
    {
        this->memorySize = memorySize;
    }

    // mengembalikan nilai atribut memorySize
    int getMemorySize()
    {
        return this->memorySize;
    }

    // mengeset nilai atribut supportsCuda
    void setSupportsCuda(string supportsCuda)
    {
        this->supportsCuda = supportsCuda;
    }

    // mengembalikan nilai atribut supportsCuda
    string getSupportsCuda()
    {
        return this->supportsCuda;
    }

    // menampilkan atribut Memory
    void printMemory()
    {
        cout << "Frequency : " << this->getFrequency() << endl;
        cout << "Memory Size : " << this->getMemorySize() << endl;
        cout << "Support Cuda : " << this->getSupportsCuda() << endl;
    }

    // destructor
    ~Memory() {}
};

File: Java/Memory.java
// kelas yang digunakan untuk mengimplementasikan Memory dan juga turunan dari
// kelas Hardware
class Memory extends Hardware {

    // private atribute dari kelas Memory
    private String frequency;
    private int memorySize;
    private String supportsCuda;

    // konstrutor
    Memory() {
    }

    // mengeset nilai atribut frequency
    public void setFrequency(String frequency) {
        this.frequency = frequency;
    }

    // mengembalikan nilai atribut frequency
    public String getFrequency() {
        return this.frequency;
    }

    // mengeset nilai atribut memorySize
    public void setMemorySize(int memorySize) {
        this.memorySize = memorySize;
    }

    // mengembalikan nilai atribut memorySize
    public int getMemorySize() {
        return this.memorySize;
    }

    // mengeset nilai atribut supportsCuda
    public void setSupportsCuda(String supportsCuda) {
        this.supportsCuda = supportsCuda;
    }

    // mengembalikan nilai atribut supportsCuda
    public String getSupportsCuda() {
        return this.supportsCuda;
    }

    // menampilkan atribut Memory
    public void printMemory() {
        System.out.println("Frequency : " + getFrequency());
        System.out.println("Memory Size : " + getMemorySize());
        System.out.println("Support Cuda : " + getSupportsCuda());
    }
}
